pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -129,6 +130,19 @@ func (rc *reconciler) syncStates() {
 	}
 }
 
+func (rc *reconciler) isExclusiveVolume(volumeSpec *volume.Spec) bool {
+	if volumeSpec.PersistentVolume == nil {
+		return false
+	}
+	// check if this volume can be attached to multiple PODs/nodes, if yes, return false
+	for _, accessMode := range volumeSpec.PersistentVolume.Spec.AccessModes {
```

Comment:
I confirmed this with @saad-ali 

The access modes in both PV and PVC are only used for matching/binding. The readOnly values in the actual volume description are used when attaching/mounting. If you have a pod with direct volume use without involving PV and PVC,  volumeSpec.PersistentVolume will be null, I think.